### ISSUE TITLE
torch.linalg_cholesky error

### ISSUE THREAD

**Issue by hjf1997** [patel-zeel/AAAI22#2]
Thanks for your kind work

I'm running you code at [nonstat_gp_cat](https://github.com/patel-zeel/AAAI22/tree/main/nonstat_gp_cat). However, the code constantly throws an error: 

> torch.linalg_cholesky: The factorization could not be completed because the input is not positive-definite (the leading minor of order 314 is not positive-definite).

It seems that you have code to reinitialize all the parameters at [gp_train.py](https://github.com/patel-zeel/AAAI22/blob/main/nonstat_gp_cat/gp_train.py):

>             # except Exception as e:
>             #     loss_val = np.inf
>             #     torch.cuda.manual_seed(rand)
>             #     rand+=1
>             #     init_vars(model)
>                 pprint('Iter', i, 'Batch', bi, 'Failed with', e, 're inited the params')
>                 # break

I also tried this for many time, but the model still cannot get the right initializations. This reinitialization of all the parameters is the only way to solve this problem? Do you have other suggestions?

Thanks in advance :)

**Comment by hjf1997** [comment]
Hi, are you here?

**Comment by patel-zeel** [comment]
Hi @hjf1997. Sorry for not replying earlier as I missed the notification. Are you trying to use `gp_train.py` for your own dataset?

**Comment by hjf1997** [comment]
Hi @patel-zeel No, I'm using the same Beijing Airquality dataset processed by your .jupyter file. 
Besides the 'not positive-definite' error, I also met some nan errors during backpropagation.
Did you also meet this error?

**Comment by patel-zeel** [comment]
Yes, we got those errors earlier but they depend on way too many things. To help you better (and to solve potential bugs in our code), would you like to explain step by step the bare minimum process you tried so that I can reproduce the error (I'd appreciate a direct code that I can execute and hopefully see the same error as you are seeing)? please also share the torch and gpytorch versions you have used.

**Comment by hjf1997** [comment]
OK, thanks for your kind help. I will first try to fix the problem and will give you feedback in a few days if there is any problem.

**Comment by hjf1997** [comment]
BTW, will you release the processing code for the London dataset?

**Comment by tiantian0612** [comment]
hello，do you meet "No module named 'nsgp'"error?  
running lib.py,there is a bug:
 from nsgp.utils.inducing_functions import f_kmeans, f_random
ModuleNotFoundError: No module named 'nsgp' 

Can you tell me how to solve it?Thank you
